avocado/core/loader.py

```diff
@@ -700,13 +700,22 @@ def ignore_broken(klass, uid, params=None):
                     return make_broken(AccessDeniedPath, test_path)
 
             # Try to resolve test ID (keep compatibility)
-            rel_path = '%s.py' % test_name
-            test_path = os.path.join(data_dir.get_test_dir(), rel_path)
+            test_path = os.path.join(data_dir.get_test_dir(), test_name)
             if os.path.exists(test_path):
                 return self._make_avocado_tests(test_path, make_broken,
-                                                subtests_filter, rel_path)
+                                                subtests_filter, test_name)
             else:
-                return make_broken(test.MissingTest, test_name)
+                if not subtests_filter and ':' in test_name:
```

Comment:
Ideally, the test filters would be handled once, say in `_discover()`, and passed along.  But this is another one of those things that can have its place on a larger cleanup and/or refactor.  This does the job.